From 38c3f8d21da90f0833578a157e22ddabf4d576bd Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@gmail.com>
Date: Sun, 12 Feb 2012 14:28:34 +0100
Subject: [PATCH 48/84] MIPS: BCM63XX: enable USB for BCM6328

The USB controller is the same as the one on BCM6368.

Signed-off-by: Jonas Gorski <jonas.gorski@gmail.com>
---
 arch/mips/bcm63xx/dev-usb-ehci.c |    2 +-
 arch/mips/bcm63xx/dev-usb-ohci.c |    3 ++-
 drivers/usb/host/ehci-bcm63xx.c  |    2 +-
 drivers/usb/host/ohci-bcm63xx.c  |    2 +-
 4 files changed, 5 insertions(+), 4 deletions(-)

--- a/arch/mips/bcm63xx/dev-usb-ehci.c
+++ b/arch/mips/bcm63xx/dev-usb-ehci.c
@@ -39,7 +39,7 @@ static struct platform_device bcm63xx_eh
 
 int __init bcm63xx_ehci_register(void)
 {
-	if (!BCMCPU_IS_6358() && !BCMCPU_IS_6368())
+	if (!BCMCPU_IS_6328() && !BCMCPU_IS_6358() && !BCMCPU_IS_6368())
 		return 0;
 
 	ehci_resources[0].start = bcm63xx_regset_address(RSET_EHCI0);
--- a/arch/mips/bcm63xx/dev-usb-ohci.c
+++ b/arch/mips/bcm63xx/dev-usb-ohci.c
@@ -39,7 +39,8 @@ static struct platform_device bcm63xx_oh
 
 int __init bcm63xx_ohci_register(void)
 {
-	if (!BCMCPU_IS_6348() && !BCMCPU_IS_6358() && !BCMCPU_IS_6368())
+	if (!BCMCPU_IS_6328() && !BCMCPU_IS_6348() && !BCMCPU_IS_6358() &&
+	    !BCMCPU_IS_6368())
 		return 0;
 
 	ohci_resources[0].start = bcm63xx_regset_address(RSET_OHCI0);
--- a/drivers/usb/host/ehci-bcm63xx.c
+++ b/drivers/usb/host/ehci-bcm63xx.c
@@ -99,7 +99,7 @@ static int __devinit ehci_hcd_bcm63xx_dr
 		bcm_rset_writel(RSET_USBH_PRIV, 0x1c0020,
 				USBH_PRIV_TEST_6358_REG);
 
-	} else if (BCMCPU_IS_6368()) {
+	} else if (BCMCPU_IS_6328() || BCMCPU_IS_6368()) {
 
 		reg = bcm_rset_readl(RSET_USBH_PRIV, USBH_PRIV_SWAP_6368_REG);
 		reg &= ~USBH_PRIV_SWAP_EHCI_DATA_MASK;
--- a/drivers/usb/host/ohci-bcm63xx.c
+++ b/drivers/usb/host/ohci-bcm63xx.c
@@ -94,7 +94,7 @@ static int __devinit ohci_hcd_bcm63xx_dr
 		bcm_rset_writel(RSET_USBH_PRIV, 0x1c0020,
 				USBH_PRIV_TEST_6358_REG);
 
-	} else if (BCMCPU_IS_6368()) {
+	} else if (BCMCPU_IS_6328() || BCMCPU_IS_6368()) {
 		reg = bcm_rset_readl(RSET_USBH_PRIV, USBH_PRIV_SWAP_6368_REG);
 		reg &= ~USBH_PRIV_SWAP_OHCI_ENDN_MASK;
 		reg |= USBH_PRIV_SWAP_OHCI_DATA_MASK;
